File: app/src/main/java/com/example/Sleepy/ui/wake/WakeViewModel.java
package com.example.Sleepy.ui.wake;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.Sleepy.R;

import java.util.Calendar;

public class WakeViewModel extends ViewModel {

    private final MutableLiveData<String> SetTimeText;
    private final MutableLiveData<String> SetWakeText;
    private final MutableLiveData<Calendar> CurTime;

    public WakeViewModel(Context context) {
        SetTimeText = new MutableLiveData<>();
        SetTimeText.setValue(context.getString(R.string.choose_time_go_to_bed));

        SetWakeText = new MutableLiveData<>();
        SetWakeText.setValue(context.getString(R.string.wake_up));

        CurTime = new MutableLiveData<>();
        CurTime.setValue(Calendar.getInstance());
    }

    public LiveData<String> getTextWake(){ return SetWakeText; }
    public LiveData<String> getText() {
        return SetTimeText;
    }
    public LiveData<Calendar> getCurTime(){ return CurTime; }

    public void setCurTime(Calendar calendar){
        CurTime.setValue(calendar);
    }
}

File: app/src/main/java/com/example/Sleepy/main/modules/wake/domain/WakeViewModel.kt
package com.example.Sleepy.main.modules.wake.domain

import android.content.Context
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData
import com.example.Sleepy.R
import java.util.*

class WakeViewModel(context: Context) : ViewModel() {
    private val timeText: MutableLiveData<String> = MutableLiveData()
    private val wakeText: MutableLiveData<String>
    private val currentTime: MutableLiveData<Calendar>
    val textWake: LiveData<String>
        get() = wakeText
    val text: LiveData<String>
        get() = timeText

    fun getCurTime(): LiveData<Calendar> {
        return currentTime
    }

    fun setCurTime(calendar: Calendar) {
        currentTime.value = calendar
    }

    init {
        timeText.value = context.getString(R.string.choose_time_go_to_bed)
        wakeText = MutableLiveData()
        wakeText.value = context.getString(R.string.wake_up)
        currentTime = MutableLiveData()
        currentTime.value = Calendar.getInstance()
    }
}